test/VaultDepositRouter.ts

```diff
@@ -0,0 +1,494 @@
+            const amount = hre.ethers.utils.parseUnits("50", 18);
+
+            await mint(mockERC20, other, amount);
+            // approve router to send ERC20 tokens in
+            await mockERC20.connect(other).approve(router.address, amount);
+            await nft.connect(user).approve(other.address, vault.address);
+            await router.connect(other).depositERC20(vault.address, mockERC20.address, amount);
+
+            expect(await mockERC20.balanceOf(vault.address)).to.equal(amount);
+
+            // Expect reporter to show correct inventory
+            expect(await reporter.verify(vault.address)).to.eq(true);
+
+            const items = await reporter.enumerate(vault.address);
+
+            expect(items.length).to.eq(1);
+            expect(items[0].tokenAddress).to.eq(mockERC20.address);
+            expect(items[0].tokenAmount).to.eq(amount);
+        });
+
+        it("should accept deposit from an ERC20 token and report inventory", async () => {
+            const { vault, mockERC20, user, router, reporter } = ctx;
+            const amount = hre.ethers.utils.parseUnits("50", 18);
+
+            await mint(mockERC20, user, amount);
+            // approve router to send ERC20 tokens in
+            await mockERC20.connect(user).approve(router.address, amount);
+            await router.connect(user).depositERC20(vault.address, mockERC20.address, amount);
+
+            expect(await mockERC20.balanceOf(vault.address)).to.equal(amount);
+
+            // Expect reporter to show correct inventory
+            expect(await reporter.verify(vault.address)).to.eq(true);
+
+            const items = await reporter.enumerate(vault.address);
+
+            expect(items.length).to.eq(1);
+            expect(items[0].tokenAddress).to.eq(mockERC20.address);
+            expect(items[0].tokenAmount).to.eq(amount);
+        });
+
+        it("should accept deposit from an ERC721 token and report inventory", async () => {
+            const { vault, mockERC721, user, router, reporter } = ctx;
+
+            const tokenId = await mintERC721(mockERC721, user);
+
+            // approve router to send ERC721 tokens in
+            await mockERC721.connect(user).approve(router.address, tokenId);
+            await router.connect(user).depositERC721(vault.address, mockERC721.address, tokenId);
+
+            expect(await mockERC721.balanceOf(vault.address)).to.equal(1);
+            expect(await mockERC721.ownerOf(tokenId)).to.equal(vault.address);
+
+            // Expect reporter to show correct inventory
+            expect(await reporter.verify(vault.address)).to.eq(true);
+
+            const items = await reporter.enumerate(vault.address);
+
+            expect(items.length).to.eq(1);
+            expect(items[0].tokenAddress).to.eq(mockERC721.address);
+            expect(items[0].tokenId).to.eq(tokenId);
+        });
+
+        it("should accept deposit from an ERC1155 token and report inventory", async () => {
```

Comment:
grammar nit: ` it should accept an ERC115 deposit` vs. `accept deposit from an ERC1155 token`.
Same issue with lines 207, 228, 272, 295, 336, 383, 453.

Reply:
Switched these all to say `should accept deposit of...` instead of `should accept deposit from...`. Was easier to bulk change